Assets/Plugins/iOS/Shopify/Checkout/Checkout.WebViewController.swift

```diff
@@ -73,10 +83,26 @@ class WebViewController: UIViewController {
 // MARK: Actions
 extension Checkout.WebViewController {
     func loadCheckoutURL() {
-        let request = URLRequest(url: url)
+        let request = URLRequest(url: startURL)
         webView.load(request)
     }
     
+    func generateErrorPage(message: String, retryURL: URL) -> URL {
+        // Instead of using webView.loadHTMLString, we generate a new HTML file and store on disk so 
+        // we can use webView.loadFileURL which allows us differentiate between an error page and a 
+        // web page. Using webView.loadHTMLString will set the url to about:blank which isn't strong enough
+        // to tell us that it's not an error page.
+        var errorHTMLString = errorPageString.replacingOccurrences(of: "<%= polaris_css %>", with: polarisCSSString)
+        errorHTMLString = errorHTMLString.replacingOccurrences(of: "<%= retry_url %>", with: lastGoodURL.absoluteString)
+        errorHTMLString = errorHTMLString.replacingOccurrences(of: "<%= message %>", with: message)
+        
+        // Write the .html file to the temporary directory.
```

Comment:
I generally find that blocks of code requiring explicit comments to describe what they're doing, can greatly benefit from being abstracted into a dedicated, single-responsibility method that also provides self-documentation. The above can be written as:
```swift
func generateErrorPage(message: String, retryURL: URL) -> URL {
    let errorHTML = self.generateErrorHTML()
    return self.logErrorHTML(errorHTML)
}
```